src/css-token-stream-value.js

```diff
@@ -0,0 +1,69 @@
+(function(internal, scope) {
+
+    function CSSTokenStreamValue(values) {
+        if (values == undefined) {
+            values = [];
+        }
+        if (!Array.isArray(values)) {
+            throw new TypeError('CSSTokenStreamValue should be an array of string or CSSVariableReferenceValue');
+        }
+        for (var i = 0; i < values.length; i++) {
+            if (typeof values[i] != 'string' && !(values[i] instanceof CSSVariableReferenceValue)) {
+                throw new TypeError("CSSTokenStreamValue's elements should be string or CSSVariableReferenceValue");
+            }
+        }
+        this.listOfReferences = values;
+    }
+
+    internal.inherit(CSSTokenStreamValue, CSSStyleValue);
+
+    CSSTokenStreamValue.prototype.size = function() {
+        return this.listOfReferences.length;
+    }
+
+    CSSTokenStreamValue.prototype.referenceAtIndex = function(index) {
+        return this.listOfReferences[index];
+    }
+
+    CSSTokenStreamValue.prototype.keys = function() {
+        var nextIndex = 0;
+        var values = this.listOfReferences;
+
+        return {
+            next: function() {
+                return nextIndex < values.length ?
+                    { done: false, value: nextIndex++ } :
+                    { done: true, value: undefined };
```

Comment:
I have changed this one to use the helper function, but I'm not sure whether using the eval() function is good or not. :cry: